### ISSUE TITLE
problem ananse network

### ISSUE THREAD

**Issue by soever1187** [vanheeringen-lab/ANANSE#26]
Hello,

While running the command "ananse network" the error AttributeError: 'float' object has no attribute 'upper' pops up. I have checked my input files and cannot find anything wrong with them. Does anyone have any suggestions to fix this problem?

![error_ananse_network](https://user-images.githubusercontent.com/74187921/103285925-a635a400-49df-11eb-8e62-ce4856444be5.png)

Kinds regards,
S. van den Oever

**Comment by siebrenf** [comment]
Hi S, I think this is a small issue. 

As a quick fix, you can add `expression.index.astype(str)` to the script. Open the network.py file, and place it above the line with the error (442). 

Let me know if it (doesn't) work!

**Comment by soever1187** [comment]
Hi,

Only adding  ``` expression.index.astype(str) ``` didn't work. So I added ``` expression.index = expression.index.astype(str) ``` to the script and this fixed the error. 

Unfortunately a new error popped up, 'ValueError: Found array with 0 sample(s) (shape=(0,)) while a minimum of 1 is required.', after one hour of running the ananse network command (see screenshot). 

![image](https://user-images.githubusercontent.com/74187921/103544692-c6aea400-4ea0-11eb-80aa-ea8a3cd0b764.png)

Kind regards,
S. van den Oever

**Comment by siebrenf** [comment]
Here my understanding of ANANSE ends unfortunately. This is what I can tell:

You start running ANANSE network from [here](https://github.com/vanheeringen-lab/ANANSE/blob/bef91e724e2911571b3d52d5adb7be371a60a0ba/ananse/network.py#L725), and it runs until here:
```
logger.info("Aggregate binding")
features = self.aggregate_binding(ddf, prom, p, weight)

logger.info("Join expression")
expression_file = self.get_expression(fin_expression, features)
```
everything going into `features` depends on your input `binding` file. So my best bet would be to check if your binding file is in the right format. An example can be found in [the docs](https://anansepy.readthedocs.io/en/master/command-line_reference/#build-gene-regulatory-network-ananse-network). Check for NAs, or other anomalies.

**Comment by simonvh** [comment]
Quick question, how come the index of the expression file is not a string? To me this suggests that the expression file is not correct as this should be the gene name. @soever1187  can you post the first few lines of your expression file?

**Comment by soever1187** [comment]
@siebrenf Thanks for the help. I checked the binding file with the example file, the only difference I can find is that the third column in the example is called "peakRPKMScale" whereas mine is called "log10_peakRPKM". 

@simonvh I checked my expression files again, if I read them in its says that the columns are strings. In total I use three expression files (n=3) that look the same, here is how my expression files look like:

![image](https://user-images.githubusercontent.com/74187921/103653154-67b26300-4f64-11eb-8b29-08b2bc0604c1.png)
